cache/compressed_secondary_cache.cc

```diff
@@ -145,6 +142,88 @@ std::string CompressedSecondaryCache::GetPrintableOptions() const {
   return ret;
 }
 
+std::unique_ptr<CompressedSecondaryCache::CacheValueChunk>
+CompressedSecondaryCache::SplitValueIntoChunks(
+    const Slice& value, const CompressionType compression_type,
+    size_t& charge) {
+  assert(!value.empty());
+  // If charge > the max size
+  const char* src_ptr = value.data();
+  size_t src_size{charge};
+  CacheAllocationPtr ptr;
+
+  std::unique_ptr<CacheValueChunk> head = std::make_unique<CacheValueChunk>();
```

Comment:
Why not have a single buffer with CacheValueChunk in the beginning followed by the value chunk, sort of like a header+payload? That would avoid the need to do extra memory allocations.

Reply:
Addressed in the latest commit.